Description: add use-system-library build flag
 Useful for things like linux distributions that want to avoid duplication
 of libraries to ease security maintenance.
Author: Joey Hess <joeyh@joeyh.name>
Date: Mon, 15 Aug 2016 15:22:07 -0400
Forwarded: https://github.com/ocharles/argon2/pull/4

---
 argon2.cabal | 31 +++++++++++++++++++------------
 1 file changed, 19 insertions(+), 12 deletions(-)

diff --git a/argon2.cabal b/argon2.cabal
index 38f1bfc..31cbf72 100644
--- a/argon2.cabal
+++ b/argon2.cabal
@@ -16,24 +16,31 @@ extra-source-files: phc-winner-argon2/src/blake2/blake2.h
                     phc-winner-argon2/src/blake2/blake2-impl.h
                     phc-winner-argon2/src/blake2/blamka-round-ref.h
 
+flag use-system-library
+  description: Link against system libargon2 library rather than embedded copy
+  default: False
+
 library
   exposed-modules:     Crypto.Argon2.FFI
                        Crypto.Argon2
   build-depends:       base >=4.8 && <4.9, bytestring, text, transformers
   hs-source-dirs:      src
   default-language:    Haskell2010
-  c-sources: phc-winner-argon2/src/argon2.c
-             phc-winner-argon2/src/argon2.h
-             phc-winner-argon2/src/core.c
-             phc-winner-argon2/src/blake2/blake2b.c
-             phc-winner-argon2/src/thread.c
-             phc-winner-argon2/src/ref.c
-             phc-winner-argon2/src/encoding.c
-             phc-winner-argon2/src/encoding.h
-             phc-winner-argon2/src/core.h
-             phc-winner-argon2/src/thread.h
-             phc-winner-argon2/src/ref.h
-  include-dirs: phc-winner-argon2/src
+  if flag(use-system-library)
+    extra-libraries: argon2
+  else
+    c-sources: phc-winner-argon2/src/argon2.c
+               phc-winner-argon2/src/argon2.h
+               phc-winner-argon2/src/core.c
+               phc-winner-argon2/src/blake2/blake2b.c
+               phc-winner-argon2/src/thread.c
+               phc-winner-argon2/src/ref.c
+               phc-winner-argon2/src/encoding.c
+               phc-winner-argon2/src/encoding.h
+               phc-winner-argon2/src/core.h
+               phc-winner-argon2/src/thread.h
+               phc-winner-argon2/src/ref.h
+    include-dirs: phc-winner-argon2/src
 
 test-suite tests
   type: exitcode-stdio-1.0
